docs/reference-guide/modules/commands/pages/command-handlers.adoc

```diff
@@ -1,249 +1,926 @@
+    }
 
-    public IssueCardCommand(String cardId, Integer amount) {
-        this.cardId = cardId;
-        this.amount = amount;
+    protected GiftCard() {
     }
-    // omitted getters, equals/hashCode, toString functions
 }
+----
+
+<1> Instance methods with `@CommandHandler` require an existing instance to be loaded.
+
+The framework automatically loads the correct instance based on the command's routing key before invoking the handler.
+
+=== Entity creation with `@EntityCreator`
+
+In event sourcing, events need to be applied to an entity instance.
+The `@EntityCreator` annotation specifies how to create that initial instance.
+
+When an entity is loaded (either to reconstitute from events or to handle a command), Axon uses `@EntityCreator` to:
+
+1. **Create an instance from an event**: If events exist in the event store, the first event is used to create the initial instance, which is then updated by applying the remaining events.
+2. **Create an empty instance**: If a command arrives targeting an entity that has no events yet, Axon checks the `@EntityCreator` methods to find one that can create an "empty instance" (typically one that takes just the identifier).
+
+There are two patterns for entity creation:
+
+**Pattern 1: Empty instance creation (identifier only)**
+
+This pattern creates an "empty" entity instance when needed.
+Axon uses this when:
+
+* A command arrives for an entity that has no events yet
+* The entity needs to exist to handle an instance command handler
+
+[source,java]
+----
+@EventSourced
+public class GiftCard {
 
-public class RedeemCardCommand {
+    private String id;
+    private boolean created;
+    private int remainingValue;
+
+    @EntityCreator // <1>
+    public GiftCard(@InjectEntityId String id) {
+        this.id = id;
+        this.created = false;
+    }
 
-    @TargetAggregateIdentifier
-    private final String cardId;
-    private final String transactionId;
-    private final Integer amount;
+    @CommandHandler // <2>
+    public void handle(IssueCardCommand cmd, EventAppender eventAppender) {
+        if (created) {
+            throw new IllegalStateException("Card already issued");
+        }
+        eventAppender.append(new CardIssuedEvent(cmd.cardId(), cmd.amount()));
+    }
 
-    public RedeemCardCommand(String cardId, String transactionId, Integer amount) {
-        this.cardId = cardId;
-        this.transactionId = transactionId;
-        this.amount = amount;
+    @EventSourcingHandler
+    private void on(CardIssuedEvent event) {
+        this.created = true;
+        this.remainingValue = event.amount();
     }
-    // omitted getters, equals/hashCode, toString functions
 }
+----
+
+<1> `@EntityCreator` on a constructor that takes only the identifier creates an empty instance.
+Axon will use this to create an instance when a command arrives for an entity with no events.
+
+<2> An instance command handler can handle the creation command on the empty instance.
```

Comment:
This is not true. We still need a separate creational command handler on a static method. That then creates the `CardIssueEvent`. On a subsequent command, the identifier from the `CardIssueEvent` is the field used to invoke the `@EntityCreator` annotated method with.